### PR TITLE
📖 Add GCP provider instructions to getting started

### LINKED ISSUE #1731: Add GCP provider instructions to docs
**User Story**

As a user, I'd like to be able to quickly grab the GCP deploy commands from the getting started page.

**Detailed Description**

Add GCP instructions based on the latest release of the gcp infrastructure provider so that we can quickly deploy capg components.

**Anything else you would like to add:**

I'll take this on, just wanted to create a new issue for it to track it.

/kind feature

### PR DESCRIPTION
**Pull request by rsmitty**
**What this PR does / why we need it**:

Adds instructions for deploying and creating clusters with the GCP provider. Will need to be backported to release-0.2 once merged.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1731

### CODE REVIEW

**Comment by rsmitty** [comment]
@vincepri If you could double-check that I caught everything needed for the GCP provider (especially since I don't usually do the kubeadm config), that'd be awesome.

File: docs/book/src/tasks/installation.md [hunk 1]
```diff
@@ -171,6 +171,33 @@ Check the [Docker provider releases](https://github.com/kubernetes-sigs/cluster-
...
+Check the [GCP provider releases](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/releases) for an up-to-date components file.
+
+```bash
+# Create the base64 encoded credentials by catting your credentials json.
+# This command uses your environment variables and encodes
+# them in a value to be stored in a Kubernetes Secret.
+export GCP_B64ENCODED_CREDENTIALS=$( cat /path/to/gcp-credentials.json | base64 | tr -d '\n' )
+
+# Create the components.
+curl -L {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-gcp" asset:"infrastructure-components.yaml" version:"0.2.x"}} \
```
Comment (vincepri) [inline docs/book/src/tasks/installation.md]: Unfortunately there is no release of CAPG that exposes these components, we might have to make folks use `make generate-examples` instead

Comment (rsmitty) [reply docs/book/src/tasks/installation.md]: What makes the most sense for now? Should we update the version above and merge the doc as-is, but not backport to release-0.2 (with the assumption that the v0.3 release of capi will publish these docs and they'll be working by that point)? Or should I update to use the make command so we can backport?

File: docs/book/src/tasks/installation.md [hunk 2]
```diff
@@ -171,6 +171,33 @@ Check the [Docker provider releases](https://github.com/kubernetes-sigs/cluster-
...
+Check the [GCP provider releases](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/releases) for an up-to-date components file.
+
+```bash
+# Create the base64 encoded credentials by catting your credentials json.
+# This command uses your environment variables and encodes
+# them in a value to be stored in a Kubernetes Secret.
+export GCP_B64ENCODED_CREDENTIALS=$( cat /path/to/gcp-credentials.json | base64 | tr -d '\n' )
+
+# Create the components.
+curl -L {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-gcp" asset:"infrastructure-components.yaml" version:"0.3.x"}} \
```
Comment (rsmitty) [inline docs/book/src/tasks/installation.md]: Went ahead and bumped this version

File: docs/book/src/tasks/installation.md (hunk 1 above)
Comment (vincepri) [reply docs/book/src/tasks/installation.md]: Let's target v1alpha3 only for CAPG, given that we're not planning to do any v1alpha2 release there. We can leave 0.3.x, but the link might be broken until we release, which is probably ok

**Pull request merged**